source/dotnet/Library/AdaptiveCards.Rendering.Html/AdaptiveCardRenderer.cs

```diff
@@ -710,7 +713,11 @@ protected static HtmlTag ChoiceSetRender(AdaptiveChoiceSetInput adaptiveChoiceSe
 
                     foreach (var choice in adaptiveChoiceSetInput.Choices)
                     {
+                        // generate a random ID to link the <label> and <input> elements together
+                        var htmlLabelId = Guid.NewGuid().ToString().Substring(0, 6);
```

Comment:
Please add a TODO comment and create a GitHub issue to track improving the ID generation. This has a few concerns,..

- Two inputs could get the same ID, yes it's unlikely but I believe by only using 6 characters of the GUID it could happen, so we would want to store the IDs and if we conflict generate a new one
- We could conflict with IDs in the parent DOM, since this is being inserted into one giant DOM

It's not a high priority concern but it's worthy of a TODO and an issue to track it